components/Grid/Timeline.tsx

```diff
@@ -0,0 +1,92 @@
+'use client';
+import React, { HTMLAttributes, useRef } from 'react';
+import {
+ useScroll, useSpring, motion, MotionStyle, 
+} from 'framer-motion';
+import { cnb } from 'cnbuilder';
+import { Grid, type GridProps } from './Grid';
+import { type GridNegativeSpacingType, type GridWidthType } from './Grid.types';
+import { marginBottoms, type MarginType } from '@/utilities/datasource';
+import * as styles from './Grid.styles';
+
+/**
+ * This component place each React node in the children in an alternating grid pattern.
+ *
+ * @component
+ *
+ * @example
+ * <Timeline startOnRight>
+ *  <Paragraph>Item 1 - Lorem Ipsum</Paragraph>
+ *  <Paragraph>Item 2 - Lorem Ipsum</Paragraph>
+ *  <Paragraph>Item 3 - Lorem Ipsum</Paragraph>
+ * </Timeline>
+ */
+
+type TimelineProps = HTMLAttributes<HTMLElement> &
+  Pick<GridProps, 'as' | 'gap' | 'pt' | 'pb' | 'py'> & {
+    startOnRight?: boolean;
+    addCenterLine?: boolean;
+    width?: GridWidthType;
+    spacing?: MarginType | GridNegativeSpacingType;
+    // inline CSS Style for grid cell
+    gridCellStyle?: MotionStyle;
+  };
+
+export const Timeline = ({
+  as,
+  startOnRight,
+  addCenterLine,
+  width = 'inset',
+  spacing,
+  pt,
+  pb,
+  py,
+  children,
+  gridCellStyle,
+  ...props
+}: TimelineProps) => {
+  // turn children into array
+  const childrenArray = React.Children.toArray(children);
+  const containerRef = useRef(null);
+  const { scrollYProgress } = useScroll({
+    target: containerRef,
+    offset: ['-500px', '60%'],
+  });
+
+  const scaleY = useSpring(scrollYProgress, {
+    stiffness: 80,
+    damping: 30,
+    restDelta: 0.001,
+  });
```

Comment:
Feel free to try it without the spring effect. For Campaign, I found that it works well for the centered line animation, but not for some other things - feel free to test out what feels most natural 😄